dropwizard-jersey/src/main/java/io/dropwizard/jersey/guava/OptionalParamConverterProvider.java

```diff
@@ -0,0 +1,72 @@
+package io.dropwizard.jersey.guava;
+
+import com.google.common.base.Function;
+import com.google.common.base.Optional;
+import org.glassfish.hk2.api.ServiceLocator;
+import org.glassfish.jersey.internal.inject.Providers;
+import org.glassfish.jersey.internal.util.ReflectionHelper;
+import org.glassfish.jersey.internal.util.collection.ClassTypePair;
+
+import javax.inject.Inject;
+import javax.inject.Singleton;
+import javax.ws.rs.ext.ParamConverter;
+import javax.ws.rs.ext.ParamConverterProvider;
+import java.lang.annotation.Annotation;
+import java.lang.reflect.Type;
+import java.util.List;
+import java.util.Set;
+
+@Singleton
+public class OptionalParamConverterProvider implements ParamConverterProvider {
+
+    private final ServiceLocator locator;
+
+    @Inject
+    public OptionalParamConverterProvider(final ServiceLocator locator) {
+        this.locator = locator;
+    }
+
+    @Override
+    public <T> ParamConverter<T> getConverter(final Class<T> rawType, final Type genericType, final Annotation[] annotations) {
+        final List<ClassTypePair> ctps = ReflectionHelper.getTypeArgumentAndClass(genericType);
+        final ClassTypePair ctp = (ctps.size() == 1) ? ctps.get(0) : null;
+        if (ctp == null || ctp.rawClass() == String.class) {
+            return new ParamConverter<T>() {
+                @Override
+                public T fromString(final String value) {
+                    return rawType.cast(Optional.fromNullable(value));
+                }
+
+                @Override
+                public String toString(final T value) throws IllegalArgumentException {
+                    return value.toString();
+                }
+            };
+        }
+        final Set<ParamConverterProvider> converterProviders = Providers.getProviders(locator, ParamConverterProvider.class);
+        for (ParamConverterProvider provider : converterProviders) {
+            @SuppressWarnings("unchecked")
```

Comment:
Does the `@SuppressWarnings` here have any effect? Can't really see what could cause a unchecked warning right now.